packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,301 @@
+   *
+   * @attribute
    */
   readonly domainName: string;
 
   /**
    * The distribution ID for this distribution.
+   *
+   * @attribute
    */
   readonly distributionId: string;
-}
\ No newline at end of file
+}
+
+/**
+ * Properties for a Distribution
+ *
+ * @experimental
+ */
+export interface DistributionProps {
+  /**
+   * The default behavior for the distribution.
+   */
+  readonly defaultBehavior: BehaviorOptions;
+
+  /**
+   * Additional behaviors for the distribution, mapped by the pathPattern that specifies which requests to apply the behavior to.
+   *
+   * @default no additional behaviors are added.
+   */
+  readonly additionalBehaviors?: Record<string, BehaviorOptions>;
+
+  /**
+   * A certificate to associate with the distribution. The certificate must be located in N. Virginia (us-east-1).
+   *
+   * @default the CloudFront wildcard certificate (*.cloudfront.net) will be used.
+   */
+  readonly certificate?: acm.ICertificate;
+
+  /**
+   * The price class that corresponds with the maximum price that you want to pay for CloudFront service.
+   * If you specify PriceClass_All, CloudFront responds to requests for your objects from all CloudFront edge locations.
+   * If you specify a price class other than PriceClass_All, CloudFront serves your objects from the CloudFront edge location
+   * that has the lowest latency among the edge locations in your price class.
+   *
+   * @default PRICE_CLASS_ALL
+   */
+  readonly priceClass?: PriceClass;
+
+  /**
+   * How CloudFront should handle requests that are not successful (e.g., PageNotFound).
+   *
+   * @default - No custom error configuration.
+   */
+  readonly errorConfigurations?: CfnDistribution.CustomErrorResponseProperty[];
+}
+
+/**
+ * A CloudFront distribution with associated origin(s) and caching behavior(s).
+ *
+ * @experimental
+ */
+export class Distribution extends Resource implements IDistribution {
+
+  /**
+   * Creates a Distribution construct that represents an external (imported) distribution.
+   */
+  public static fromDistributionAttributes(scope: Construct, id: string, attrs: DistributionAttributes): IDistribution {
+    return new class extends Resource implements IDistribution {
+      public readonly domainName: string;
+      public readonly distributionId: string;
+
+      constructor() {
+        super(scope, id);
+        this.domainName = attrs.domainName;
+        this.distributionId = attrs.distributionId;
+      }
+    }();
+  }
+
+  /**
+   * Creates a Distribution for an S3 Bucket, where the bucket has not been configured for website hosting.
+   *
+   * This creates a single-origin distribution with a single default behavior.
+   */
+  public static forBucket(scope: Construct, id: string, bucket: s3.IBucket): Distribution {
+    return new Distribution(scope, id, {
+      defaultBehavior: { origin: Origin.fromBucket(scope, 'SingleOriginBucket', bucket) },
```

Comment:
This seems strange - why not `Origin.fromBucket(bucket)`?

Reply:
The scope and id are necessary because under the hood the S3Origin creates an OriginAccessIdentity (a Resource) and grants it read access to the bucket. The alternative -- which I considered but can't remember exactly why I discarded -- would be to have the Origin expose a bind() method (or similar), and do the OAI + grants in that, rather than at creation.

Reply:
bind() pattern is our go-to for such cases.

As a rule of thumb, `fromXxx` methods should not accept `scope` and `id`.